ansible/library/generate_checksum_file.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/python
+
+from ansible.module_utils.basic import *
+import sys, os, hashlib, json, yaml
+
+# Calculate md5 hash and return
+def md5Checksum(filePath):
+    with open(filePath, 'rb') as fh:
+        m = hashlib.md5()
+        while True:
+            data = fh.read(8192)
+            if not data:
+                break
+            m.update(data)
+        return m.hexdigest()
+
+# Calculate sha256 hash and return
+def sha256Checksum(filePath):
+    with open(filePath, 'rb') as fh:
+        contents = fh.read()
+        m = hashlib.sha256(contents)
+    return m.hexdigest()
+
+# Calculate sha512 hash and return
+def sha512Checksum(filePath):
+    with open(filePath, 'rb') as fh:
+        contents = fh.read()
+        m = hashlib.sha512(contents)
+    return m.hexdigest()
+
+def checkPathExists(filePath):
+    if os.path.exists(filePath):
+        return True
+    else:
+        return False
+
+def generateChecksumData(checksumType, source):
+    if checksumType == 'all':
+        return { 'md5': md5Checksum(source), 'sha256': sha256Checksum(source), 'sha512': sha512Checksum(source) }
+    elif checksumType == 'md5':
+        return { 'md5': md5Checksum(source) }
+    elif checksumType == 'sha256':
+        return { 'sha256': sha256Checksum(source) }
+    else:
+        return { 'sha512': sha512Checksum(source) }
+
+def toJSONFile(outData, outFile):
+    with open(outFile, 'w') as out:
+        json.dump(outData, out, indent=4)
+
+def toYAMLFile(outData, outFile):
+    with open(outFile, 'w') as out:
+        out.write('---\n')
+        yaml.safe_dump(outData, out, default_flow_style=False)
+
+def main():
+    module = AnsibleModule(
+      argument_spec = dict(
+        src             = dict(required=True, type='str'),
+        dest_path       = dict(required=True, type='str'),
+        dest_file       = dict(required=True, type='str'),
+        outfile_type    = dict(required=False, choices=['json', 'yaml'], default='json'),
+        checksum_type   = dict(required=False, choices=['all', 'md5', 'sha256', 'sha512'], default='all'),
+      )
+    )
```

Comment:
@cliffano Module options here. Includes output to either JSON or YAML file types, and generating md5, sha256, sha512 or all of the above hashes.